docs/decisions/0001-meilisearch.rst

```diff
@@ -0,0 +1,146 @@
+
+Open edX Search API
+===================
+
+We are proposing a new abstraction package with more generalised architecture inorder to support all available search engines including elasticsearch, meilisearch and algolia.
+The Open edX Search API is designed with a forward-thinking approach, ensuring seamless integration with a wide range of future search engines.
+This package aims to provide flexibility, scalability, and ease of use for developers looking to enhance their search capabilities within the Open edX platform.
+
+
+Architecture
+============
+
+The Open edX Search API comprises four main components: the Driver, Indexer, Load Indexes Command, and a JavaScript library. Each component is described in detail below.
+
+Driver Class
+============
+
+The base driver class contains the following functions: `get_user_token()`, `check_connection()`, `indexes()`, `index()`, and `get_search_rules()`. To integrate a new search engine or driver, the child class must implement all these functions.
+This class is responsible for initializing the communication protocol between client and the search service. To change the driver class by updating the value of `SEARCH_ENGINE` in django settings.
+the default value is `SEARCH_ENGINE = "openedx_search_api.drivers.meilisearch.MeiliSearchEngine"`
+
+Indexer Class
+=============
+
+The base indexer class consists of the `index()` and `index_documents()` functions. To override the default functionality of the base class, the child indexer class should implement these functions.
+Inorder to update the indexer class set `INDEXER_CLASS` attribute in django settings default value is `INDEXER_CLASS = "openedx_search_api.indexers.base.BaseIndexer"`
+
+Load Indexes
+============
+
+A management command reads index configurations from the Django settings module and populates the indexes accordingly. further more we can also add configurable parameters to update specific index and set specific dataset.
+
+`python ./manage.py load_indexes`
+
+Javascript SDK
+==============
+
+A global JavaScript client SDK is available to facilitate communication with search services, serving as an abstraction layer to form search engine queries. Once the package is installed on the backend, this SDK will be accessible via a public static endpoint. By default, it supports Meilisearch, but it is structured to be adaptable for use with other search services.
+
+The SearchEngine object includes the following functions:
+
+1. `queryBuilder`
+2. `getSearchURL`
+3. `search`
+4. `request`
+
+Examples
+========
+Below is an example of adding a client SDK:
+```html
+<script src="<%= process.env.BASE_URL %>/static/django_search_backend/js/search_library.js" type="text/javascript"></script>
+```
+
+We have created an example to showcase this [here](https://github.com/openedx/frontend-app-learning/compare/master...qasimgulzar:frontend-app-learning:qasim/autosuggest-courseware).
```

Comment:
I'm not sure why, but this link and the following one are not rendering properly in GitHub's markdown preview.

Reply:
I will check

Reply:
fixed [here](https://github.com/qasimgulzar/openedx-search-api/pull/5)